styles/scss/core/_utilities.scss

```diff
@@ -21,13 +21,17 @@
 $color-levels: 100, 200, 300, 400, 500, 600, 700, 800, 900;
 
 .x-small {
-  font-size: $x-small-font-size;
+  font-size: $x-small-font-size !important;
```

Comment:
I'm always slightly wary of `!important`, could you explain why it's needed here? (Depending on the reasoning it might be nice to add that as a comment in the file itself)

Reply:
CSS utility class definitions generally _should_ always have `!important`. See list of other [existing utility classes](https://paragon-openedx.netlify.app/foundations/css-utilities/) on the Paragon docs site to observe their inclusion of `!important`.

See example(s) of CSS utility class definitions from Bootstrap v4 ([source](https://github.com/twbs/bootstrap/blob/e5643aaa89eb67327a5b4abe7db976f0ea276b70/scss/utilities/_sizing.scss#L11-L20)) as a reference, too.

Reply:
Ah yes, I was getting towards the end of the diff and missed the filename.

It looks like *almost* everything else in the file has `!important` (only `.icon-spin` doesn't) - is the lack of `!important` in `.icon-spin` intentional?

Reply:
Hmm, in additional `icon-spin`, the `font-weight: normal` property on `.micro` doesn't include `!important` either, though I don't it should for `font-weight` (you could want micro-sized copy with `font-weight-bold`, etc.). IMHO, `.micro` shouldn't be defining `font-weight` at all.

Regarding `.icon-spin` specifically, it appears to only be used with the `StatefulButton` component across all of the `openedx` GitHub organization. I actually wonder whether we truly want to consider this a global utility class or not, given its specific to icons.

For example, alternatively, should the `Icon` component expose a `spin` prop that extends the rendered icon to add the `icon-spin` class conditionally?

E.g., proposed usage from `StatefulButton`:

```jsx
icons: {
  pending: <Icon src={SpinnerSimple} spin />,
},
```

I'm struggling to identify use cases for `.icon-spin` outside of the `Icon` component. With `icon-spin` as a global CS utility class, I could also see it getting abused to make things spin/animate on the page that shouldn't (e.g., non-icons).

My 2c recommendation: 
* Remove `.icon-spin` as a CSS utility class, implementing it instead of a class name specific to `Icon` component.
* Expose optional `spin` prop on `Icon` component.
* Update `StatefulButton` to pass `spin` prop on its `pending` icon.

Reply:
> My 2c recommendation:
>
> * Remove .icon-spin as a CSS utility class, implementing it instead of a class name specific to Icon component.
> * Expose optional spin prop on Icon component.
> * Update StatefulButton to pass spin prop on its pending icon.

This absolutely sounds like the right move to me!

> IMHO, `.micro` shouldn't be defining `font-weight` at all.

Do we have a sense of what might break if it *didn't*?

Reply:
That being said, I don't think either of those should be part of this PR. Just making a couple follow-up issues to address them seems like the right move to me.

Reply:
> > IMHO, .micro shouldn't be defining font-weight at all.
>
> Do we have a sense of what might break if it didn't?

It shouldn't break anything. The font-weight in `.micro` is font-weight: normal`, which corresponds to font-weight 400, the default `body` font weight used elsewhere.

> That being said, I don't think either of those should be part of this PR. Just making a couple follow-up issues to address them seems like the right move to me.

100% 😄